src/components/Marketplace/MarketplaceScreen.js

```diff
@@ -30,61 +33,166 @@ const INITIAL_POPUP_PRODUCE = {
-          const groupPerPallet = record.fields['grouped produce type per pallet'] || 0;
-          let palletPrice = pricePerGroup * groupPerPallet;
-          let hasAgencyPrice = false;
-          const userRole = getInitialRole();
-          if (userRole === 'agency' && agencyPricePerGroup < pricePerGroup) {
-            hasAgencyPrice = true;
-            palletPrice = agencyPricePerGroup * groupPerPallet;
-          }
-          // Capitalise 1st letter of produce type to match filtering
-          let produceType = record.fields['produce category'] ? record.fields['produce category'][0] : 'No category';
-          produceType = produceType.charAt(0).toUpperCase() + produceType.slice(1);
-          const recordInfo = {
-            listingID: record.fields['listing id'],
-            produceID: record.fields.produce ? record.fields.produce[0] : -1,
-            farmID: record.fields['farm id'],
-            palletPrice,
-            season: record.fields['growing season'] || 'No season',
-            produceType,
-            palletsAvailable: record.fields['pallets available'] || 0,
-            hasAgencyPrice,
-          };
-          processedProduceRecords.push(recordInfo);
-        });
-        setProduceListings(processedProduceRecords);
+  // helper fx to async check produce farm name
+  const searchProduceFarmName = (produce) => {
+    const { farmID } = produce;
+    let containsTerm = false;
+
+    if (farmID === undefined || farmID[0].length < 5) { return false; }
+
+    const farmArr = farmID.toString().split(',');
+    return new Promise((resolve) => {
+      base('Farms').find(farmArr[0]).then((farmObj) => {
+        if (checkFieldForSearchTerm(farmObj.fields['farm name']) === true) { containsTerm = true; }
+        return resolve(containsTerm);
       });
+    });
+  };
+
+  // helper fx to check individual produce for name or farm name match to search term
+  async function searchProduce(produce) {
+    return await searchProduceName(produce) || searchProduceFarmName(produce);
+  }
+
+  // filter produce by name/farm match to user search terms
+  async function filterProducebySearch(listings, setListings) {
+    const data = Array.from(listings);
+    Promise.all(data.map((produce) => searchProduce(produce)))
+    // Use the result of the promises to filter the produceListings
+      .then((result) => setListings(data.filter((element, index) => result[index])));
+  }
+
+  // run on submit of search bar (search icon or enter clicked)
+  // may be useful to add any side bar filtering methods here too
+  const filterBySearch = () => {
+    filterProducebySearch(allProduceListings, setProduceListings);
+    if (shopByFarmID === undefined) {
+      filterFarmsBySearch();
+    }
+    // else the shopByFarm method is called and will filter after fetching records
+  };
+
+  // Get prod id, grouped unit type, price per grouped unit for each of a farm's produce records
+  // run if shopping by farm
```

Comment:
What does this comment mean?

Reply:
I meant that this method grabs all the produce record info for a given farmID and that it's called when shopping by farm, but the comment is pretty unclear so I'll edit it! if you have any preferences for how it should be phrased lmk!